clang/lib/AST/FormatString.cpp

```diff
@@ -810,6 +811,7 @@ bool FormatSpecifier::hasValidLengthModifier(const TargetInfo &Target,
         case ConversionSpecifier::sArg:
         case ConversionSpecifier::ScanListArg:
         case ConversionSpecifier::ZArg:
+        case ConversionSpecifier::CHERIpArg:
```

Comment:
This should probably be restricted to hybrid/only allow for `l`?

Reply:
This is within `LengthModifier::AsLong`?

Reply:
Hmm, I had assumed it would be hard to restrict this to hybrid, but since TargetInfo is present perhaps I can make that work.  Would need to figure out how to make the tests DTRT.

Reply:
It depends whether that's what we want. Having to use:

- `%p` for non-CHERI
- `%#lp` for hybrid
- `%#p` for purecap

would be rather annoying; permitting a redundant `l` in `%#lp` for purecap would help (and there's not much more you can do to help that given no other compiler/libc will let you write `%#lp`... well, some might ignore the characters and let you get away with it, but they don't have to).

Reply:
Hmm, I'm fine permitting 'lp' for purecap.  I guess it's akin to how we permit `__capability` in purecap as well.